citizenvoice/apiapp/views.py

```diff
@@ -6,150 +6,229 @@
     @staticmethod
-    def GetOrderedQuestionFromSurvey(survey_id, question_order):
-        queryset = SurveyViewSet.GetSurvey(survey_id)[0].question_set.all().filter(order=question_order)
-        serializer_class = QuestionSerializer
+    def GetQuestionBySurvey(survey_id): 
+        """
+        Get specific Questions based on its survey_id.
+
+        Parameters:
+            survey_id (int): Survey ID to be used for finding related Questions.
+
+        Return: 
+            queryset: containing the Question instance related to this survey
+        """
+
+        queryset = Question.objects.filter(survey=survey_id)
+        return queryset  
+
+    @staticmethod
+    def GetOrderedQuestionBySurvey(survey_id, question_order):
+        """
+        Get specific Questions based on its survey_id, and a specific order.
+
+        Parameters:
+            survey_id (int): Survey ID to be used for finding related Questions.
+            question_order (int): The order in which the questions in the Survey are to be displayed.
+
+        Return: 
+            queryset: containing the Question instance related to this survey, of a given order
+        """
+        queryset = Question.objects.filter(survey=survey_id, order=question_order)
         return queryset
+        
 
-# Create a ViewSet that queries all the instances of Survey in the database, and parse them through the serializer
 class SurveyViewSet(viewsets.ModelViewSet):
     """
-    Survey ViewSet used internally to query data from database. The following functions are defined in this viewset:
+    Survey ViewSet used internally to query data from database.
 
-    GetSurveys() - returns a set of all Survey instances in the database
+    """
 
-    GetSurvey(int id) - returns an instance of Survey with the ID that was provided in the function.
+    serializer_class = SurveySerializer
 
-    """
+    def get_queryset(response):
+        """
+        Returns a set of all Survey instances in the database.
+
+        Return:
+            queryset: containing all Survey instances
+        """
 
-    # Get all surveys
-    @staticmethod
-    def GetSurveys():
         queryset = Survey.objects.all().order_by('name')
-        serializer_class = SurveySerializer
         return queryset
 
-    # Get a specific Survey based on its ID
     @staticmethod
-    def GetSurvey(id):
+    def GetSurveyByID(id):
+        """
+        Get a specific Survey based on its ID.
+
+        Parameters:
+            id (int): Survey ID to be used for finding this Survey.
+
+        Return: 
+            queryset: containing the Survey instance with this id
+        """
         queryset = Survey.objects.filter(id=id)
-        serializer_class = SurveySerializer
         return queryset
 
-# Create a ViewSet that queries all the instances of Response in the database, and parse them through the serializer
+    @staticmethod
+    def GetSurveyByAuthor(author):
```

Comment:
For consistency, please rename the class author to **designer**, and also all related functions

Reply:
Ah yeah, I forgot this one! Changing now.